README.md

```diff
@@ -1,35 +1,90 @@
 # cvs-svc-technical-records
+Please install and configure [nvm](https://github.com/nvm-sh/nvm) with this node version so it can be used during installation.
+
+
+## Architecture
+
+To capture and persist technical records for a Vehicle submitted by the client (VTM) which is then made available to VTA/VTM for searching a vehicle. If the submitted technical record is applicable for a ministry plate certificate, it triggers the generation of the certificate via dynamo streams of the technical results table.
+
+### End to end design
+[All in one view](https://wiki.dvsacloud.uk/pages/viewpage.action?pageId=79254695)
+
+![All in one](/docs/allinone.png)
+
+### Technical record microservice
+[Further details about this micro service](https://wiki.dvsacloud.uk/pages/viewpage.action?spaceKey=HVT&title=Technical+Records+Microservice).
+
+#### API Specs
+[Technical records API](https://wiki.dvsacloud.uk/download/attachments/33863060/API_Vehicle_Tech_Records_v32.yaml?version=2&modificationDate=1591883253725&api=v2).
+
+
+#### Database Model
+Being a NoSQL database data isn't stored in the relational model, but as a JSON document which is native to dynamodb database. The JSON document is the representation of API specification, hence API specification should be referred, to understand document structure stored in the database.
+
+#### Database Indexes
+       Table Name: cvs-{$environment-name}-technical-records
+
+       Partition / Hash Key: systemNumber
+
+       Sort Key: vin
+
+       LSI: N/A
+
+       GSI:
+              primaryVrm (Partition Key),
+
+              trailerId (Partition Key),
+
+              vin (Partition Key),
+
+              partialVin (Partition Key),
+
+#### Interactions/Integration with other AWS service
+The diagram illustrates that technical records integrate with Test Number service to request a unique number (systemNumber, trailorId, plateSerialNumber), before the information is saved into dynamo table. Technical record service also receives status updates from test result microservice to update a record from Provisional to Active [refer Test Result Submisson triggering Technical Record Status update](https://wiki.dvsacloud.uk/display/HVT/Test+Result+Submisson+triggering+Technical+Record+Status+update).
+![Integration AWS](/docs/awsintegration.png)
+
+
```

Comment:
include following lines here "Above Diagram shows orchestration/event publishing involved within Technical Records service. Following are the approved patterns it uses:"

Reply:
done